### PR TITLE
Updated cses-2079 with python code.

### PR DESCRIPTION
**Pull request by ravitejaBZ**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/cses-2079.mdx [hunk 1]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
 ```
 
 </JavaSection>
+	
+<PySection>
+
+```py
+	
+maxn = 200005
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:134]: rename to MAX_N

File: solutions/cses-2079.mdx [hunk 2]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
...
+g = [[] for i in range(maxn)] # graph
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:135]: two spaces after comment
like
```suggestion
g = [[] for i in range(maxn)]  # graph
```

File: solutions/cses-2079.mdx [hunk 3]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
...
+s = [0 for i in range(maxn)] # size of subtree
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:136]: just name this `subtree_size`

File: solutions/cses-2079.mdx [hunk 4]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
...
+
+def dfs_size(cur: int, par: int) -> None:
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:138]: what does this do?

File: solutions/cses-2079.mdx [hunk 5]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
...
+	# size of a subtree is the sum of the sizes of all child subtrees + 1.
+	s[cur] = 1
+	for chi in g[cur]:
+		if chi != par:
+			dfs_size(chi, cur)
+			s[cur] += s[chi]
+
+def get_centroid(cur: int,  par: int) -> int:
+	'''
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:147]: use double quotes

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cses-2079.mdx [hunk 6]
```diff
@@ -131,26 +131,26 @@ public class Centroid {
 
 ```py
 	
-maxn = 200005
-g = [[] for i in range(maxn)]  # graph
-s = [0 for i in range(maxn)] # size of subtree
+MAX_N = 200005
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:134]: is the +5 necessary?

File: solutions/cses-2079.mdx [hunk 7]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
 ```
 
 </JavaSection>
+	
+<PySection>
+
+```py
+	
+MAX_N = 200001
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:134]: why +1?

Comment (ravitejaBZ) [reply solutions/cses-2079.mdx:134]: We will be getting Runtime error if we dont do that.  We are filling the values in g and subtree_size in 1 index based way

Comment (SansPapyrus683) [reply solutions/cses-2079.mdx:134]: can you make the values 0-indexed then?

Comment (ravitejaBZ) [reply solutions/cses-2079.mdx:134]: In general such problems are solved using 1 index based for better understanding. that is why the commited code that are present in Java and C++ are done in the same way.

Comment (SansPapyrus683) [reply solutions/cses-2079.mdx:134]: debatable
in that case remove the + 1 and just use `MAX_N + 1` everywhere else instead

File: solutions/cses-2079.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/cses-2079.mdx:138]: unaddressed

Comment (ravitejaBZ) [reply solutions/cses-2079.mdx:138]: https://github.com/cpinitiative/usaco-guide/pull/3075/commits/30723be695212664dc78e397c527270309f424a0 

Please check the latest commit

Comment (SansPapyrus683) [reply solutions/cses-2079.mdx:138]: yeah? it doesn't put a comment saying what `dfs_size` does

Comment (ravitejaBZ) [reply solutions/cses-2079.mdx:138]: added it below line 138

Comment (SansPapyrus683) [reply solutions/cses-2079.mdx:138]: that...doesn't tell me what the function does
i'd add a doc comment saying "calculates all subtree sizes"

Comment (ravitejaBZ) [reply solutions/cses-2079.mdx:138]: updated it

File: solutions/cses-2079.mdx [hunk 8]
```diff
@@ -126,5 +126,43 @@ public class Centroid {
...
+			dfs_size(chi, cur)
+			subtree_size[cur] += subtree_size[chi]
+
+def get_centroid(cur: int,  par: int) -> int:
+	"""
+	if possible, move towards the adjacent node with the subtree size
+	greater than n / 2. Otherwise, we have found our centroid.
+	"""
+	for chi in g[cur]:
+		if chi != par:
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:151]: use the names `child`, `parent`, and `curr` instead of `chi`, `par`, & `cur`

**Review by SansPapyrus683 — APPROVED** [review]
@envyaims

**Review by envyaims — APPROVED** [review]
gr8

**Pull request merged**